### PR TITLE
add Dropwizard metrics5

### PR DESCRIPTION
**Pull request by sullis**
new module: spectator-reg-metrics5

### CODE REVIEW

File: spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+      return buf.toString();
+    });
+  }
+
+  /** Create a new instance. */
+  public MetricsRegistry(Clock clock, io.dropwizard.metrics5.MetricRegistry impl, Function<Id, String> formatNameFn) {
+    super(clock);
+    this.impl = impl;
+    this.registeredGauges = new ConcurrentHashMap<>();
+    this.formatNameFn = formatNameFn;
```
Comment (brharrington) [inline spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java:62]: I don't think there is a need for a variant that takes a formatting function, with the `metrics5.MetricName` it should just have the tags and let the reporter figure out how to send it to other sources.

File: spectator-reg-metrics4/build.gradle [hunk 2]
```diff
@@ -0,0 +1,13 @@
+dependencies {
+  api project(':spectator-api')
+  api 'io.dropwizard.metrics:metrics-core:4.2.8'
```
[the remark this answers is not in the document]
Comment (sullis) [reply spectator-reg-metrics4/build.gradle]: I have removed spectator-reg-metrics4

File: spectator-reg-metrics4/src/main/java/com/netflix/spectator/metrics4/DoubleGauge.java [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2014-2019 Netflix, Inc.
```
[the remark this answers is not in the document]
Comment (sullis) [reply spectator-reg-metrics4/src/main/java/com/netflix/spectator/metrics4/DoubleGauge.java]: Fixed.

File: spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+  /** Create a new instance. */
+  public MetricsRegistry(Clock clock, io.dropwizard.metrics5.MetricRegistry impl) {
+    this(clock, impl, id -> {
+      Id normalized = Utils.normalize(id);
+      StringBuilder buf = new StringBuilder();
+      buf.append(normalized.name());
+      for (Tag t : normalized.tags()) {
+        buf.append('.').append(t.key()).append('-').append(t.value());
+      }
+      return buf.toString();
```
[the remark this answers is not in the document]
Comment (sullis) [reply spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java]: Fixed.

File: spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java (hunk 1 above)
Comment (sullis) [reply spectator-reg-metrics5/src/main/java/com/netflix/spectator/metrics5/MetricsRegistry.java]: Fixed.

**Comment by sullis** [comment]
ready for review @brharrington

**Review by brharrington — APPROVED** [review]

**Pull request merged**